### DISCUSSION TITLE
Welcome to mad-migration Discussions!

### DISCUSSION THREAD

**Discussion by AliyevH** [MadeByMads/mad-migration#37] [Ideas]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by maestro-1** [comment]
Good day
This looks like a truly great tool, however, I would like to know in what ways this is different from ansible, as I believe this knowledge will be key.

**Reply by Turall** [reply]
Good day
Welcome to our project. Why are you comparing this tool with Ansible? Ansible is not designed for database migration. MadMigration helps people migrate their data from one database to another. This tool does not do some kind of SQL dump, it migrates the data as you specify.

**Reply by Turall** [reply]
For example, you have a table with columns containing first name, last name, nickname, age, and birthday with type string in a MySQL database. Some programmers have created the table structure at this stage before you. And now you notice that this is wrong, the date of birth should not be a string, etc. And for some reason, you want to switch to PostgreSQL. Here's a tool to fix it all. :) You specify the database and structure in the YAML file and MadMigration will do it for you and migrate all data.

**Reply by maestro-1** [reply]
@Turall Thank you for your answer, I was not comparing it to ansible, I just needed to fully grasp the difference because I initially thought ansible had such a feature as well. Again thank you for clearing up my confusion.

**Reply by Turall** [reply]
Yes, Ansible has this feature, but it does it with backups and we don't use this approach because we can rename database names, column names, types, or split one column and migrate it on a new table by 3 columns.
